src/Http/Cookie/Cookie.php

```diff
@@ -0,0 +1,506 @@
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Validates the cookie name
+     *
+     * @param string $name Name of the cookie
+     * @return void
+     * @throws \InvalidArgumentException
+     */
+    protected function validateName($name)
+    {
+        if (preg_match("/[=,; \t\r\n\013\014]/", $name)) {
+            throw new InvalidArgumentException(
+                sprintf('The cookie name `%s` contains invalid characters.', $name)
+            );
+        }
+
+        if (empty($name)) {
+            throw new InvalidArgumentException('The cookie name cannot be empty.');
+        }
+    }
+
+    /**
+     * Gets the raw cookie value
+     *
+     * @return string
+     */
+    public function getValue()
+    {
+        return $this->value;
+    }
+
+    /**
+     * Sets the raw cookie data
+     *
+     * @param string|array $value Value of the cookie to set
+     * @return $this
+     */
+    public function setValue($value)
+    {
+        if (is_array($value)) {
+            $this->isExpanded = true;
+        }
+
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * Sets the path
+     *
+     * @param string|null $path Sets the path
+     * @return $this
+     */
+    public function setPath($path)
+    {
+        $this->path = $path;
+
+        return $this;
+    }
+
+    /**
+     * Sets the domain
+     *
+     * @param string $domain Domain to set
+     * @return $this
+     */
+    public function setDomain($domain)
+    {
+        $this->domain = $domain;
+
+        return $this;
+    }
+
+    /**
+     * Check if the cookie is secure
+     *
+     * @return bool
+     */
+    public function isSecure()
+    {
+        return $this->secure;
+    }
+
+    /**
+     * Set HTTP Only
+     *
+     * @param bool $httpOnly HTTP Only
+     * @return $this
+     */
+    public function setHttpOnly($httpOnly)
+    {
+        if (!is_bool($httpOnly)) {
+            throw new InvalidArgumentException(sprintf(
+                'The provided arg must be of type `bool` but `%s` given',
+                gettype($httpOnly)
+            ));
+        }
+
+        $this->httpOnly = $httpOnly;
+
+        return $this;
+    }
+
+    /**
+     * Check if the cookie is HTTP only
+     *
+     * @return bool
+     */
+    public function isHttpOnly()
+    {
+        return $this->isHttpOnly();
+    }
+
+    /**
+     * Sets the expiration date
+     *
+     * @param \DateTimeInterface $dateTime Date time object
+     * @return $this
+     */
+    public function expiresAt(DateTimeInterface $dateTime)
+    {
+        $this->expiresAt = (int)$dateTime->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Checks if a value exists in the cookie data
+     *
+     * @param string $path Path to check
+     * @return bool
+     */
+    public function check($path)
+    {
+        return Hash::check($this->value, $path);
```

Comment:
You do this check in [read()](https://github.com/cakephp/cakephp/pull/10203/files/b452e4cddd4491a51d5b7076efebe1734382aee2#diff-8139d3fada6116a8baa223443e8825b2R355)

I thought you could do it as well here.